src/main/java/org/snomed/snowstorm/rest/MultiSearchController.java

```diff
@@ -75,7 +78,71 @@ public ItemsPage<BrowserDescriptionSearchResult> findDescriptions(
+		for (String branchPath : branchConceptIds.keySet()) {
+			conceptMiniMap.putAll(conceptService.findConceptMinis(branchPath, branchConceptIds.get(branchPath), languageDialects).getResultsMap());
+			timer.checkpoint("Join concepts from " + branchPath);
+		}
+
+		List<BrowserDescriptionSearchResult> results = new ArrayList<>();
+		for (Description description : descriptions) {
+			BrowserDescriptionSearchResult result = new BrowserDescriptionSearchResult(description.getTerm(), description.isActive(), description.getLanguageCode(),
+					description.getModuleId(), conceptMiniMap.get(description.getConceptId()));
+			result.addExtraField("branchPath", description.getPath());
+			results.add(result);
+		}
+		timer.finish();
+
+		return new ItemsPage<>(new PageImpl<>(results, pageRequest, descriptions.getTotalElements()));
+	}
+	
+	@Operation(description = "Search descriptions across multiple Code Systems. Can search specified branches in addition to all published branches.")
+	@PostMapping(value = "multisearch/descriptions", produces = {"application/json", "text/csv"})
+	@JsonView(value = View.Component.class)
+	public ItemsPage<BrowserDescriptionSearchResult> findDescriptionsIncludingRequestedBranches(
+			@RequestParam String term,// Required!
+			@RequestParam(required = false) Boolean active,
+			@RequestParam(required = false) Collection<String> module,
+			@RequestParam(required = false) String ecl,
+
+			@Parameter(description = "Set of two character language codes to match. " +
+					"The English language code 'en' will not be added automatically, in contrast to the Accept-Language header which always includes it. " +
+					"Accept-Language header still controls result FSN and PT language selection.")
+			@RequestParam(required = false) Set<String> language,
+
+			@Parameter(name = "Set of description types to include. Pick descendants of '900000000000446008 | Description type (core metadata concept) |'.")
+			@RequestParam(required = false) Set<Long> type,
+
+			@RequestParam(required = false) Boolean conceptActive,
+			@RequestParam(defaultValue = "ALL_PUBLISHED_CONTENT") ContentScope contentScope,
+			@RequestParam(defaultValue = "0") int offset,
+			@RequestParam(defaultValue = "50") int limit,
+			@RequestBody MultibranchDescriptionSearchRequest includeBranches,
+			@RequestHeader(value = "Accept-Language", defaultValue = Config.DEFAULT_ACCEPT_LANG_HEADER) String acceptLanguageHeader) throws TooCostlyException {
+
+		TimerUtil timer = new TimerUtil("MultiSearch - Descriptions");
+
+		DescriptionCriteria descriptionCriteria = new DescriptionCriteria()
+				.term(term)
+				.active(active)
+				.modules(module)
+				.searchLanguageCodes(language)
+				.type(type)
+				.conceptActive(conceptActive);
+
+		PageRequest pageRequest = ControllerHelper.getPageRequest(offset, limit);
+		Page<Description> descriptions = multiSearchService.findDescriptions(descriptionCriteria, ecl, pageRequest, includeBranches.getBranches());
```

Comment:
This particular call was developed along with @pgwilliams, and was reviewed and approved by him in PR #489.  If at all possible it would be good not to make major changes now, as the current code has been tested and validated on the integration branch.  Is this confusing enough to warrant re-working at this point, or is the current state useable?